homeassistant/components/pcal9535a/switch.py

```diff
@@ -0,0 +1,103 @@
+"""Support for switch sensor using I2C PCAL9535A chip."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.switch import PLATFORM_SCHEMA
+from homeassistant.const import DEVICE_DEFAULT_NAME
+from homeassistant.helpers.entity import ToggleEntity
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_INVERT_LOGIC = "invert_logic"
+CONF_I2C_ADDRESS = "i2c_address"
+CONF_I2C_BUS = "i2c_bus"
+CONF_PINS = "pins"
+CONF_STRENGTH = "strength"
+
+STRENGTH_025 = "0.25"
+STRENGTH_050 = "0.5"
+STRENGTH_075 = "0.75"
+STRENGTH_100 = "1.0"
+
+DEFAULT_INVERT_LOGIC = False
+DEFAULT_I2C_ADDRESS = 0x20
+DEFAULT_I2C_BUS = 1
+DEFAULT_STRENGTH = STRENGTH_100
+
+_SWITCHES_SCHEMA = vol.Schema({cv.positive_int: cv.string})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_PINS): _SWITCHES_SCHEMA,
+        vol.Optional(CONF_INVERT_LOGIC, default=DEFAULT_INVERT_LOGIC): cv.boolean,
+        vol.Optional(CONF_STRENGTH, default=DEFAULT_STRENGTH): vol.In(
+            [STRENGTH_025, STRENGTH_050, STRENGTH_075, STRENGTH_100]
+        ),
+        vol.Optional(CONF_I2C_ADDRESS, default=DEFAULT_I2C_ADDRESS): vol.Coerce(int),
+        vol.Optional(CONF_I2C_BUS, default=DEFAULT_I2C_BUS): cv.positive_int,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the PCAL9535A devices."""
+    from pcal9535a import PCAL9535A
+
+    invert_logic = config.get(CONF_INVERT_LOGIC)
```

Comment:
Please use `dict[key]` for required config keys and keys with default config schema values.